content/en/blog/_posts/2022-12-16-statefulset-migration.md

```diff
@@ -0,0 +1,212 @@
+---
+layout: blog
+title: "Kubernetes 1.26: StatefulSet Migration"
+date: 2022-12-16
+slug: statefulset-migration
+---
+
+**Author**: Peter Schuurman (Google)
+
+Kubernetes v1.26 introduces a new, alpha-level feature for
+[StatefulSets](/docs/concepts/workloads/controllers/statefulset/) that controls
+the ordinal numbering of Pod replicas. Ordinals can start from arbitrary
+non-negative numbers. This blog post will discuss how this feature can be
+used.
+
+## Background
+
+StatefulSets ordinals provide sequential identities for pod replicas. When using
+[OrderedReady Pod Management](/docs/tutorials/stateful-application/basic-stateful-set/#orderedready-pod-management),
+Pods are created from ordinal index `0` up to `N-1`.
+
+With Kubernetes today, orchestrating a StatefulSet migration across clusters is
+challenging. Backup and restore solutions exist, but these require the
+application to be scaled down to zero replicas prior to migration. In today's
+fully connected world, planned downtime and unavailability may not allow you to
+meet your business goals. You could use
+[Cascading Delete](/docs/tutorials/stateful-application/basic-stateful-set/#cascading-delete)
+or
+[On Delete](/docs/tutorials/stateful-application/basic-stateful-set/#on-delete)
+to migrate individual pods, however this is error prone and tedious to manage.
+You lose the self-healing benefit of the StatefulSet controller when your Pods
+fail or are evicted.
+
+This feature enables a StatefulSet to be responsible for a range of ordinals
+within a logical range of `[0, N)`. With it, you can scale down a range
+(`[0, k)`) in a source cluster, and scale up the complementary range (`[k, N)`)
```

Comment:
You might need to explain the notation here for readers who are not mathematicians.

Reply:
Updated with an example, and a reference to half-open interval